packages/esm-patient-list-management-app/src/list-details-table/list-details-table.test.tsx

```diff
@@ -1,6 +1,29 @@
 import React from 'react';
-import { render, screen } from '@testing-library/react';
+import { render, screen, fireEvent, waitFor } from '@testing-library/react';
 import ListDetailsTable from './list-details-table.component';
+import { showSnackbar } from '@openmrs/esm-framework';
+
+jest.mock('@openmrs/esm-framework', () => {
+  const actual = jest.requireActual('@openmrs/esm-framework');
+  return {
+    ...actual,
+    showSnackbar: jest.fn(),
+    useLayoutType: () => 'desktop',
+    isDesktop: () => true,
+    ExtensionSlot: ({ state }) => (
+      <button onClick={() => state.selectPatientAction('new-patient')}>{state.buttonText}</button>
+    ),
+  };
+});
```

Comment:
```suggestion
const mockShowSnackbar = jest.mocked(showSnackbar);
const mockUseLayoutType = jest.mocked(useLayoutType);
const mockIsDesktop = jest.mocked(isDesktop);
const mockExtensionSlot = jest.mocked(ExtensionSlot);

beforeEach(() => {
  mockUseLayoutType.mockReturnValue('small-desktop');
  mockIsDesktop.mockReturnValue(true);
  mockShowSnackbar.mockImplementation(() => {});
  mockExtensionSlot.mockImplementation(({ state }: any) => (
    <button onClick={() => state.selectPatientAction('new-patient')}>{state.buttonText}</button>
  ));
});
```

[We shouldn't need to use partial mocks](https://o3-docs.openmrs.org/docs/frontend-modules/unit-and-integration-testing.en-US#you-should-not-need-to-use-partial-mocks).